### PR TITLE
Adjusted Stardew YAML to force progressive goals on based on slot's goal.

### PR DESCRIPTION
**Pull request by ploki122**
Added triggers to ensure that the most egregious cases don't happen (like rolling craftmaster without craftsanity checks). This makes it so working toward your goal will explicitly send checks.

### CODE REVIEW

File: games/Stardew Valley.yaml [hunk 1]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
   gifting: 'true'
   mods: []
   death_link: 'false'
+triggers: #Added triggers to complement the goal (eg. add museumsanity to a museum goal)
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: community_center
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:137]: Suggestion: if CC goal is rolled, vanilla bundle odds should be transferred to either remixed or thematic. Vanilla CC is boring as a goal I think

File: games/Stardew Valley.yaml [hunk 2]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            normal: 25
+            expensive: 5
+            very_expensive: 5
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: master_angler #catch all fishsanity fish
+      options:
+        Stardew Valley:
+          fishsanity: #Move none odds to all
+            all: 10
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:152]: I would also move the "only easy fish" odds to all, for a total of 30-20-50

File: games/Stardew Valley.yaml [hunk 3]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            exclude_legendaries: 20
+            exclude_hard_fish: 50
+            only_easy_fish: 20
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: complete_collection #finish museum
+      options:
+        Stardew Valley:
+          museumsanity: #remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:161]: Can we scale that to 75-25, just to maintain consistency with the rest of the yaml where odds are built as a percentage?

File: games/Stardew Valley.yaml [hunk 4]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            milestones: 30
+            all: 10
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: full_shipment # ship all items, follows shipsanity
+      options:
+        Stardew Valley:
+          shipsanity: #Remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:169]: Scaling to 100

Comment (agilbert1412) [inline games/Stardew Valley.yaml:169]: Can we split this a bit, to maybe include some fish options too? I'm still against the "everything" setting for a yaml, but the other ones would bring a bit of variety

File: games/Stardew Valley.yaml [hunk 5]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            crops: 15
+            full_shipment: 5
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: gourmet_chef # cook every recipe, follows cooksanity
+      options:
+        Stardew Valley:
+          cooksanity: #Remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:177]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 6]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            queen_of_sauce: 20
+            all: 10
+          chefsanity: #Increase the odds shuffling recipes to add
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:180]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 7]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            none: 10
+            all: 10
+            random: 10
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: craft_master #craft every item
+      options:
+        Stardew Valley:
+          craftsanity: #Remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:189]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 8]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            all: 20
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: protector_of_the_valley # complete monsterslayer goals, follows monstersanity
+      options: 
+        Stardew Valley: #Remove one_per_category odds
+          monstersanity:
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:196]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 9]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            very_short_goals: 20
+            short_goals: 20
+            progressive_goals: 5
+            split_goals: 5
+    - option_category: Stardew Valley
+      option_name: goal
+      option_result: mystery_of_the_stardrops #find every stardrop
+      options:
+        Stardew Valley:
+          museumsanity: #Remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:207]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 10]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            milestones: 30
+            all: 10
+          festival_locations: #Remove disable odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:210]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 11]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            easy: 40
+            hard: 40
+          friendsanity: #Remove none odds
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:213]: Scaling to 100

File: games/Stardew Valley.yaml [hunk 12]
```diff
@@ -131,3 +131,88 @@ Stardew Valley:
...
+            bachelors: 25
+            starting_npcs: 25
+            all: 10
+          fishsanity: #Set to all, since that is required
```
Comment (agilbert1412) [inline games/Stardew Valley.yaml:217]: Scaling to 100

Comment (agilbert1412) [inline games/Stardew Valley.yaml:217]: Fishsanity doesn't need to be "all" to shuffle the stardrop. The stardrop is quite literally the only item generated from fishsanity, so no matter how many locations are added by it, the stardrop always shuffles. You can keep odds for other settings pretty much the same, as long as it's not "none"

**Review by agilbert1412 — CHANGES_REQUESTED** [review]

File: games/Stardew Valley.yaml (hunk 1 above)
Comment (ploki122) [reply games/Stardew Valley.yaml:137]: I feel like vanilla bundles should either be removed entirely from the YAML, or that CC goal should have them. Either the vanilla bundle is fun or it's not, and I don't think Vanilla is meaningfully shorter or less varied than remixed/thematic.

File: games/Stardew Valley.yaml (hunk 4 above)
Comment (ploki122) [reply games/Stardew Valley.yaml:168]: My opinion is the same for this and CC : If it's worth changing shipsanity for full_shipment, it's probably worth changing it for the entire YAML. In this case, however, I prefer changing it for the entire YAML.

**Review by agilbert1412 — APPROVED** [review]

**Pull request merged**